src/main/java/teammates/ui/template/CourseEditInstructorPanel.java

```diff
@@ -55,28 +57,45 @@ public CourseEditInstructorPanel(int instructorToShowIndex, int instructorIndex,
-        List<CourseEditSectionRow> blankRows = new ArrayList<CourseEditSectionRow>();
+        Map<Integer, String> specialSectionNames = new TreeMap<Integer, String>();
+        Map<Integer, String> nonSpecialSectionNames = new TreeMap<Integer, String>();
+        
         int sectionIndex = -1;
         for (String sectionName : sectionNames) {
             sectionIndex++;
-            CourseEditSectionRow sectionRow = new CourseEditSectionRow(sectionName, sectionNames,
-                                                                       sectionIndex, instructor,
-                                                                       instructorIndex, feedbackNames);
-            
-            // break section rows into special and blank lists
-            if (sectionRow.isSectionSpecial()) {
-                rows.add(sectionRow);
+            if (isSectionSpecial(sectionName)) {
+                specialSectionNames.put(sectionIndex, sectionName);
             } else {
-                blankRows.add(sectionRow);
-                if (firstBlankSectionRowIndex == sectionNames.size()) {
-                    firstBlankSectionRowIndex = sectionRow.getSectionIndex();
-                }
+                nonSpecialSectionNames.put(sectionIndex, sectionName);
             }
         }
         
-        // append blank list to the special list, so all special rows appear before blank rows
-        for (CourseEditSectionRow blankRow : blankRows) {
-            rows.add(blankRow);
+        int panelIndex = -1;
+        for (Map.Entry<Integer, String> sectionNameEntry : specialSectionNames.entrySet()) {
+            panelIndex++;
+            sectionIndex = sectionNameEntry.getKey();
+            String sectionName = sectionNameEntry.getValue();
+            CourseEditSectionRow sectionRow = new CourseEditSectionRow(sectionName, sectionNames, sectionIndex,
+                                                                       panelIndex, instructor,
+                                                                       instructorIndex, feedbackNames);
+            
+            rows.add(sectionRow);
+        }
+        
+        for (Map.Entry<Integer, String> sectionNameEntry : nonSpecialSectionNames.entrySet()) {
+            panelIndex++;
+            
+            if (firstBlankSectionRowIndex == sectionNames.size()) {
+                firstBlankSectionRowIndex = panelIndex;
+            }
+            
+            sectionIndex = sectionNameEntry.getKey();
+            String sectionName = sectionNameEntry.getValue();
+            CourseEditSectionRow sectionRow = new CourseEditSectionRow(sectionName, sectionNames, sectionIndex,
+                                                                       panelIndex, instructor,
+                                                                       instructorIndex, feedbackNames);
+            
+            rows.add(sectionRow);
         }
         return rows;
     }
```

Comment:
This method is 50 lines long with many loops. Any possibility of shortening it using SLAP?